reactive/kotlinx-coroutines-reactor/src/Flux.kt

```diff
@@ -32,9 +42,238 @@ import kotlin.coroutines.*
  * **Note: This is an experimental api.** Behaviour of publishers that work as children in a parent scope with respect
  *        to cancellation and error handling may change in the future.
  */
+
 @ExperimentalCoroutinesApi
 fun <T> CoroutineScope.flux(
     context: CoroutineContext = EmptyCoroutineContext,
     @BuilderInference block: suspend ProducerScope<T>.() -> Unit
-): Flux<T> =
-    Flux.from(publish(newCoroutineContext(context), block = block))
+): Flux<T> = Flux.from(reactorPublish(newCoroutineContext(context), block = block))
```

Comment:
It looks like instead of having `is CoreSubscriber` hack (that may stop working at some point, it looks like we are relying on implementation detail) we should use `Flux.create {}` API here in the same way we are using `Mono.create {}`

Reply:
And this is where we should abstract some parts of `PublisherCoroutine` away to share most of the implementation details